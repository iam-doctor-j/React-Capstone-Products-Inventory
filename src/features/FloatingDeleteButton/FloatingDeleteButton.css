.circle {
    border-radius: 50%;
    box-shadow: 0 0.1rem 0.3rem rgba(0,0,0,0.5);
    padding: 1rem;
}
.floating-bottom {
    position: fixed;
    bottom: 10%;
    right: 10%;
    z-index: 10000;
}
.far {
    height: 0.5rem;
    width: 1.5rem;
}

/* @media only screen and (max-width: 550px) {
    .floating-bottom {
        left: 75%;
    }
}
@media only screen and (min-width: 550px) and (max-width: 850px) {
    .floating-bottom {
        left: 85%;
    }
} */
