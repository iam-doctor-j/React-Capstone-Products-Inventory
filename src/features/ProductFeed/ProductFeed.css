.menu-list {
    list-style: none;
}
.card {
    border: none;
    border-radius: .15rem;
    margin: 1rem 0;
    box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
}
.card-subtitle {
    font-size: 0.8rem;
}
.add-button {
    height: 2rem;
    width: 2rem;
    position: relative;
    opacity: 0.5;
}
.flex-center-child {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

hr {
    margin-top: 0px;
    margin-bottom: 0px;
}
.product-link, .product-link:hover, .product-link:active, .product-link:visited {
    color: inherit;
    text-decoration: none;
}
