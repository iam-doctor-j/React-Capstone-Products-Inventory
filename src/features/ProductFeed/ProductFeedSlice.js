import { createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import * as _ from 'lodash';

export const productFeedSlice = createSlice({
  name: 'products',
  initialState: {
    products: [],
    deleteList: [],
  },
  reducers: {
    addProduct: (state, action) => {
      state.products.push(action.payload);
    },
    editProduct: (state, action) => {
      let index = _.findIndex(
        state.products,
        (prod) => prod.id === action.payload.id
      );
      state.products.splice(index, 1, action.payload);
    },
    fetchProducts: (state, action) => {
      state.products = action.payload;
    },
    viewProduct: (state, action) => {
      let product = action.payload;
      let index = _.findIndex(state.products, (prod) => prod.id === product.id);
      state.products[index].views = product.views;
    },
    deleteProducts: (state, action) => {
        let l = action.payload.length;
        for(let i=0;i<l;i++){
            _.remove(state.products, prod => prod.id === action.payload[i].id);
        }
    },
    addProductToDeleteList: (state, action) => {
        console.warn(state);
        state.deleteList.push(action.payload);
    },
    removeProductToDeleteList: (state, action) => {
        let index = _.findIndex(state.deleteList, prod => prod.id === action.payload);
        state.deleteList.splice(index, 1);
    },
    onSearchProducts: (state, action) => {
      let result = _.filter(state.products, prod => prod.name.value.includes(action.payload));
      state.products = result;
    }
  },
});

const {
  addProduct,
  editProduct,
  fetchProducts,
  viewProduct,
  deleteProducts,
  addProductToDeleteList,
  removeProductToDeleteList,
  onSearchProducts,
} = productFeedSlice.actions;

export const searchProducts = (name) => dispatch => {
  dispatch(onSearchProducts(name));
}

export const addProductToDb = (product, cb) => (dispatch) => {
  axios
    .post('http://localhost:4000/products', product)
    .then((res) => {
      dispatch(addProduct(res.data));
      cb();
    })
    .catch((err) => {
      throw err;
    });
};

export const editProductToDb = (id, product, success, error) => (dispatch) => {
  axios
    .put('http://localhost:4000/products/' + id, product)
    .then((res) => {
      dispatch(editProduct(res.data));
      success();
    })
    .catch((err) => {
      throw err;
    });
};

export const fetchProductsFromDb = () => (dispatch) => {
  axios
    .get('http://localhost:4000/products')
    .then((res) => dispatch(fetchProducts(res.data)))
    .catch((err) => {
      throw err;
    });
};

export const viewProductFromDb = (id) => (dispatch) => {
  axios.get('http://localhost:4000/products/' + id).then((res) => {

    let views = +res.data.views;
    axios
      .patch('http://localhost:4000/products/' + id, { views: ++views })
      .then((res) => {
        dispatch(viewProduct(res.data));
      })
      .catch((err) => {
        throw err;
      });
  });
};

export const addProductToDeleteListFirst = (product) => dispatch => {
    dispatch(addProductToDeleteList(product));
};

export const removeProductToDeleteListFirst = (id) => dispatch => {
    dispatch(removeProductToDeleteList(id));
}

export const deleteProductsFromDb = (products, success) => dispatch => {
    let toDelete = products.map(prod => {
        return axios.delete('http://localhost:4000/products/'+prod.id);
    });
    dispatch(deleteProducts(products));
    success && success();
};

export const productSelector = (state) => state.products;

export default productFeedSlice.reducer;