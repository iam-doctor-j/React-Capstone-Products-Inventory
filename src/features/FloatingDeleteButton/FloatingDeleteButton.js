import React from "react";
import './FloatingDeleteButton.css'
import { deleteProductsFromDb } from "../ProductFeed/ProductFeedSlice";
import routes from "../../constants/routes";
import { toast } from "react-toastify";

export default function FloatingDeleteButton(props) {
    return(
        <button className="btn btn-danger circle floating-bottom" onClick={ () => {
            if(props.loggedIn) {
                console.log(props.deleteList)
                props.dispatch(deleteProductsFromDb(props.deleteList, () => toast.error('Product(s) deleted!')))
            } 
            else
                props.history.push(routes.LOGIN);
            } }>
            <i className="far fa-trash-alt"></i>
        </button>
    );
}