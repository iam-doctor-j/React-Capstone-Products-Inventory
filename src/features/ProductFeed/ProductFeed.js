import React from 'react';
import { Link } from 'react-router-dom';
import Card from 'react-bootstrap/Card'
import Plus from '../../assets/svgs/solid/plus.svg';
import './ProductFeed.css';
import * as _ from 'lodash';
import { connect } from 'react-redux';
import { fetchProductsFromDb, addProductToDeleteListFirst, removeProductToDeleteListFirst, searchProducts } from './ProductFeedSlice';
import Product from './Product/Product';
import FloatingDeleteButton from '../FloatingDeleteButton/FloatingDeleteButton';
import routes from '../../constants/routes';

class ProductFeed extends React.Component {
    constructor(props) {
        super(props);
        this.searchText = React.createRef();
    }
    
    componentDidMount() {
        if(this.props.products.length === 0)
            this.props.dispatch(fetchProductsFromDb());
    }

    addToDeleteList = (product) => {
        this.props.dispatch(addProductToDeleteListFirst(product));
    }

    removeFromDeleteList = (id) => {
        this.props.dispatch(removeProductToDeleteListFirst(id));
    }

    isInDeleteList = (id) => {
        if(this.props.deleteList.length === 0)
            return false;
        let arr = _.find(this.props.deleteList, prod => prod.id === id);
        if(!arr)
            return false;
        else 
            return true;
    } 

    handleSearchOver = ({target}) => {
        if(target.value === '') {
            this.props.dispatch(fetchProductsFromDb());
        }
    }

    render() {
        
        let productList = this.props.products && this.props.products.map((product, i) => {
            return(
                <li key={product.id}>
                    <Product {
                        ...{
                            ...this.props, 
                            addToDeleteList: this.addToDeleteList, 
                            removeFromDeleteList: this.removeFromDeleteList,
                            isInDeleteList: this.isInDeleteList(product.id)
                        }
                    } 
                    index={i}
                    />
                </li>
            );
        });
        return(
            <div className="container h-100">
                <FloatingDeleteButton {...this.props}/>
            
                <div className="row mt-3">
                    <div className="col-12 col-md-6">
                    <Link title="Add New Product" to="/add">
                        <Card>
                            <Card.Body className="flex-center-child">
                                <img src={Plus} alt="Add Product" className="add-button"/>
                            </Card.Body>
                        </Card>
                    </Link>
                    </div>
                    <div className="col-12 col-md-6">
                    <Card>
                        <Card.Body className="form-inline d-flex">
                            <input type="text" className="form-control search-box flex-grow-1" ref={this.searchText} placeholder="search products" onChange={this.handleSearchOver}/>
                            <button className="btn btn-link text-secondary" onClick={() => this.props.dispatch(searchProducts(this.searchText.current.value))}><i className="fas fa-search"></i></button>
                        </Card.Body>
                    </Card>
                    </div>
                </div>
                <div className="row mt-3">
                    <div className="col-12">
                        <Card>
                            <Card.Body>
                                <Link to={routes.CHART} className="btn btn-outline-secondary w-100 label-text">
                                    top products chart
                                    <i className="far fa-chart-bar ml-1"></i>
                                </Link>
                            </Card.Body>
                        </Card>
                    </div>
                </div>
                <div className="heading-text mt-3">All products</div>
                <div>
                    { productList.length !== 0  && 
                        <ul style={{listStyle: 'none', padding: 0}}>
                            { productList}
                        </ul>
                    }
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        products: state.productFeed.products,
        loggedIn: state.auth.loggedIn,
        deleteList: state.productFeed.deleteList,
    }
}

export default connect(mapStateToProps)(ProductFeed);

// export default ProductFeed;