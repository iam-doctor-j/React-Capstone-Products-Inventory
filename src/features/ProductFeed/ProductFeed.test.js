import React from 'react';
import { shallow } from 'enzyme';
import {ProductFeed} from './ProductFeed';
import FloatingDeleteButton from '../FloatingDeleteButton/FloatingDeleteButton';

describe('Test ProductFeed with shallow rendering', () => {
    let wrapper;

    let strProds = `{
        "products": [
        {
            "id": "63796e43-f23b-542e-99a7-11db520c7bb6",
            "views": 6,
            "name": {
              "value": "Laptop",
              "visible": true
            },
            "description": {
              "value": "A simple Laptop",
              "visible": true
            },
            "manufacturer": {
              "value": "Lenovo",
              "visible": true
            },
            "price": {
              "value": 64999,
              "visible": true
            },
            "quantity": {
              "value": 1,
              "visible": true
            }
          },
          {
            "id": "d8548a39-7a6a-5258-bf6c-92fde0dc21fa",
            "views": 3,
            "name": {
              "value": "Mobile Phone",
              "visible": true
            },
            "description": {
              "value": "Smartphone",
              "visible": true
            },
            "manufacturer": {
              "value": "Realme",
              "visible": true
            },
            "price": {
              "value": 15000,
              "visible": true
            },
            "quantity": {
              "value": 1,
              "visible": true
            }
          },
          {
            "id": "cc61beb5-3315-5a82-9b34-da193d64a55c",
            "views": 4,
            "name": {
              "value": "Computer",
              "visible": true
            },
            "description": {
              "value": "A desktop PC fully assembled with all components. Mouse and Keyboard included.",
              "visible": true
            },
            "manufacturer": {
              "value": "HP",
              "visible": true
            },
            "price": {
              "value": 35000,
              "visible": true
            },
            "quantity": {
              "value": 1,
              "visible": false
            }
          },
          {
            "id": "32d9fd8f-91dd-5668-b003-83020ea5439f",
            "views": 4,
            "name": {
              "value": "Radio",
              "visible": true
            },
            "description": {
              "value": "A old radio",
              "visible": true
            },
            "manufacturer": {
              "value": "Saregama",
              "visible": true
            },
            "price": {
              "value": 1200,
              "visible": true
            },
            "quantity": {
              "value": 10,
              "visible": true
            }
        }
        ]
    }`;

    let props = JSON.parse(strProds);

    props = {
        ...props,
        loggedIn: true,
        deleteList: [],
        dispatch: () => {}
    }

    beforeEach(() => {
        wrapper = shallow(<ProductFeed {...props}/>);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a FloatingDeleteButton', () => {
        expect(wrapper.find(FloatingDeleteButton).length).toEqual(1);
    })

    afterEach(() => {
        wrapper.unmount();
    })
});