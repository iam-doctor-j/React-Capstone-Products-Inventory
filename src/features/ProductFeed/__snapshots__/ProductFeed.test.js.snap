// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProductFeed with shallow rendering should render correctly 1`] = `
<div
  className="container h-100"
>
  <FloatingDeleteButton
    deleteList={Array []}
    dispatch={[Function]}
    loggedIn={true}
    products={
      Array [
        Object {
          "description": Object {
            "value": "A simple Laptop",
            "visible": true,
          },
          "id": "63796e43-f23b-542e-99a7-11db520c7bb6",
          "manufacturer": Object {
            "value": "Lenovo",
            "visible": true,
          },
          "name": Object {
            "value": "Laptop",
            "visible": true,
          },
          "price": Object {
            "value": 64999,
            "visible": true,
          },
          "quantity": Object {
            "value": 1,
            "visible": true,
          },
          "views": 6,
        },
        Object {
          "description": Object {
            "value": "Smartphone",
            "visible": true,
          },
          "id": "d8548a39-7a6a-5258-bf6c-92fde0dc21fa",
          "manufacturer": Object {
            "value": "Realme",
            "visible": true,
          },
          "name": Object {
            "value": "Mobile Phone",
            "visible": true,
          },
          "price": Object {
            "value": 15000,
            "visible": true,
          },
          "quantity": Object {
            "value": 1,
            "visible": true,
          },
          "views": 3,
        },
        Object {
          "description": Object {
            "value": "A desktop PC fully assembled with all components. Mouse and Keyboard included.",
            "visible": true,
          },
          "id": "cc61beb5-3315-5a82-9b34-da193d64a55c",
          "manufacturer": Object {
            "value": "HP",
            "visible": true,
          },
          "name": Object {
            "value": "Computer",
            "visible": true,
          },
          "price": Object {
            "value": 35000,
            "visible": true,
          },
          "quantity": Object {
            "value": 1,
            "visible": false,
          },
          "views": 4,
        },
        Object {
          "description": Object {
            "value": "A old radio",
            "visible": true,
          },
          "id": "32d9fd8f-91dd-5668-b003-83020ea5439f",
          "manufacturer": Object {
            "value": "Saregama",
            "visible": true,
          },
          "name": Object {
            "value": "Radio",
            "visible": true,
          },
          "price": Object {
            "value": 1200,
            "visible": true,
          },
          "quantity": Object {
            "value": 10,
            "visible": true,
          },
          "views": 4,
        },
      ]
    }
  />
  <div
    className="row mt-3"
  >
    <div
      className="col-12 col-md-6"
    >
      <Link
        title="Add New Product"
        to="/add"
      >
        <Card
          body={false}
        >
          <CardBody
            className="flex-center-child"
          >
            <img
              alt="Add Product"
              className="add-button"
              src="plus.svg"
            />
          </CardBody>
        </Card>
      </Link>
    </div>
    <div
      className="col-12 col-md-6"
    >
      <Card
        body={false}
      >
        <CardBody
          className="form-inline d-flex"
        >
          <input
            className="form-control search-box flex-grow-1"
            onChange={[Function]}
            placeholder="search products"
            type="text"
          />
          <button
            className="btn btn-link text-secondary"
            onClick={[Function]}
          >
            <i
              className="fas fa-search"
            />
          </button>
        </CardBody>
      </Card>
    </div>
  </div>
  <div
    className="row mt-3"
  >
    <div
      className="col-12"
    >
      <Card
        body={false}
      >
        <CardBody>
          <Link
            className="btn btn-outline-secondary w-100 label-text"
            to="/chart"
          >
            top products chart
            <i
              className="far fa-chart-bar ml-1"
            />
          </Link>
        </CardBody>
      </Card>
    </div>
  </div>
  <div
    className="heading-text mt-3"
  >
    All products
  </div>
  <div>
    <ul
      style={
        Object {
          "listStyle": "none",
          "padding": 0,
        }
      }
    >
      <li
        key="63796e43-f23b-542e-99a7-11db520c7bb6"
      >
        <Product
          addToDeleteList={[Function]}
          deleteList={Array []}
          dispatch={[Function]}
          index={0}
          isInDeleteList={false}
          loggedIn={true}
          products={
            Array [
              Object {
                "description": Object {
                  "value": "A simple Laptop",
                  "visible": true,
                },
                "id": "63796e43-f23b-542e-99a7-11db520c7bb6",
                "manufacturer": Object {
                  "value": "Lenovo",
                  "visible": true,
                },
                "name": Object {
                  "value": "Laptop",
                  "visible": true,
                },
                "price": Object {
                  "value": 64999,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": true,
                },
                "views": 6,
              },
              Object {
                "description": Object {
                  "value": "Smartphone",
                  "visible": true,
                },
                "id": "d8548a39-7a6a-5258-bf6c-92fde0dc21fa",
                "manufacturer": Object {
                  "value": "Realme",
                  "visible": true,
                },
                "name": Object {
                  "value": "Mobile Phone",
                  "visible": true,
                },
                "price": Object {
                  "value": 15000,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": true,
                },
                "views": 3,
              },
              Object {
                "description": Object {
                  "value": "A desktop PC fully assembled with all components. Mouse and Keyboard included.",
                  "visible": true,
                },
                "id": "cc61beb5-3315-5a82-9b34-da193d64a55c",
                "manufacturer": Object {
                  "value": "HP",
                  "visible": true,
                },
                "name": Object {
                  "value": "Computer",
                  "visible": true,
                },
                "price": Object {
                  "value": 35000,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": false,
                },
                "views": 4,
              },
              Object {
                "description": Object {
                  "value": "A old radio",
                  "visible": true,
                },
                "id": "32d9fd8f-91dd-5668-b003-83020ea5439f",
                "manufacturer": Object {
                  "value": "Saregama",
                  "visible": true,
                },
                "name": Object {
                  "value": "Radio",
                  "visible": true,
                },
                "price": Object {
                  "value": 1200,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 10,
                  "visible": true,
                },
                "views": 4,
              },
            ]
          }
          removeFromDeleteList={[Function]}
        />
      </li>
      <li
        key="d8548a39-7a6a-5258-bf6c-92fde0dc21fa"
      >
        <Product
          addToDeleteList={[Function]}
          deleteList={Array []}
          dispatch={[Function]}
          index={1}
          isInDeleteList={false}
          loggedIn={true}
          products={
            Array [
              Object {
                "description": Object {
                  "value": "A simple Laptop",
                  "visible": true,
                },
                "id": "63796e43-f23b-542e-99a7-11db520c7bb6",
                "manufacturer": Object {
                  "value": "Lenovo",
                  "visible": true,
                },
                "name": Object {
                  "value": "Laptop",
                  "visible": true,
                },
                "price": Object {
                  "value": 64999,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": true,
                },
                "views": 6,
              },
              Object {
                "description": Object {
                  "value": "Smartphone",
                  "visible": true,
                },
                "id": "d8548a39-7a6a-5258-bf6c-92fde0dc21fa",
                "manufacturer": Object {
                  "value": "Realme",
                  "visible": true,
                },
                "name": Object {
                  "value": "Mobile Phone",
                  "visible": true,
                },
                "price": Object {
                  "value": 15000,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": true,
                },
                "views": 3,
              },
              Object {
                "description": Object {
                  "value": "A desktop PC fully assembled with all components. Mouse and Keyboard included.",
                  "visible": true,
                },
                "id": "cc61beb5-3315-5a82-9b34-da193d64a55c",
                "manufacturer": Object {
                  "value": "HP",
                  "visible": true,
                },
                "name": Object {
                  "value": "Computer",
                  "visible": true,
                },
                "price": Object {
                  "value": 35000,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": false,
                },
                "views": 4,
              },
              Object {
                "description": Object {
                  "value": "A old radio",
                  "visible": true,
                },
                "id": "32d9fd8f-91dd-5668-b003-83020ea5439f",
                "manufacturer": Object {
                  "value": "Saregama",
                  "visible": true,
                },
                "name": Object {
                  "value": "Radio",
                  "visible": true,
                },
                "price": Object {
                  "value": 1200,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 10,
                  "visible": true,
                },
                "views": 4,
              },
            ]
          }
          removeFromDeleteList={[Function]}
        />
      </li>
      <li
        key="cc61beb5-3315-5a82-9b34-da193d64a55c"
      >
        <Product
          addToDeleteList={[Function]}
          deleteList={Array []}
          dispatch={[Function]}
          index={2}
          isInDeleteList={false}
          loggedIn={true}
          products={
            Array [
              Object {
                "description": Object {
                  "value": "A simple Laptop",
                  "visible": true,
                },
                "id": "63796e43-f23b-542e-99a7-11db520c7bb6",
                "manufacturer": Object {
                  "value": "Lenovo",
                  "visible": true,
                },
                "name": Object {
                  "value": "Laptop",
                  "visible": true,
                },
                "price": Object {
                  "value": 64999,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": true,
                },
                "views": 6,
              },
              Object {
                "description": Object {
                  "value": "Smartphone",
                  "visible": true,
                },
                "id": "d8548a39-7a6a-5258-bf6c-92fde0dc21fa",
                "manufacturer": Object {
                  "value": "Realme",
                  "visible": true,
                },
                "name": Object {
                  "value": "Mobile Phone",
                  "visible": true,
                },
                "price": Object {
                  "value": 15000,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": true,
                },
                "views": 3,
              },
              Object {
                "description": Object {
                  "value": "A desktop PC fully assembled with all components. Mouse and Keyboard included.",
                  "visible": true,
                },
                "id": "cc61beb5-3315-5a82-9b34-da193d64a55c",
                "manufacturer": Object {
                  "value": "HP",
                  "visible": true,
                },
                "name": Object {
                  "value": "Computer",
                  "visible": true,
                },
                "price": Object {
                  "value": 35000,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": false,
                },
                "views": 4,
              },
              Object {
                "description": Object {
                  "value": "A old radio",
                  "visible": true,
                },
                "id": "32d9fd8f-91dd-5668-b003-83020ea5439f",
                "manufacturer": Object {
                  "value": "Saregama",
                  "visible": true,
                },
                "name": Object {
                  "value": "Radio",
                  "visible": true,
                },
                "price": Object {
                  "value": 1200,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 10,
                  "visible": true,
                },
                "views": 4,
              },
            ]
          }
          removeFromDeleteList={[Function]}
        />
      </li>
      <li
        key="32d9fd8f-91dd-5668-b003-83020ea5439f"
      >
        <Product
          addToDeleteList={[Function]}
          deleteList={Array []}
          dispatch={[Function]}
          index={3}
          isInDeleteList={false}
          loggedIn={true}
          products={
            Array [
              Object {
                "description": Object {
                  "value": "A simple Laptop",
                  "visible": true,
                },
                "id": "63796e43-f23b-542e-99a7-11db520c7bb6",
                "manufacturer": Object {
                  "value": "Lenovo",
                  "visible": true,
                },
                "name": Object {
                  "value": "Laptop",
                  "visible": true,
                },
                "price": Object {
                  "value": 64999,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": true,
                },
                "views": 6,
              },
              Object {
                "description": Object {
                  "value": "Smartphone",
                  "visible": true,
                },
                "id": "d8548a39-7a6a-5258-bf6c-92fde0dc21fa",
                "manufacturer": Object {
                  "value": "Realme",
                  "visible": true,
                },
                "name": Object {
                  "value": "Mobile Phone",
                  "visible": true,
                },
                "price": Object {
                  "value": 15000,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": true,
                },
                "views": 3,
              },
              Object {
                "description": Object {
                  "value": "A desktop PC fully assembled with all components. Mouse and Keyboard included.",
                  "visible": true,
                },
                "id": "cc61beb5-3315-5a82-9b34-da193d64a55c",
                "manufacturer": Object {
                  "value": "HP",
                  "visible": true,
                },
                "name": Object {
                  "value": "Computer",
                  "visible": true,
                },
                "price": Object {
                  "value": 35000,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 1,
                  "visible": false,
                },
                "views": 4,
              },
              Object {
                "description": Object {
                  "value": "A old radio",
                  "visible": true,
                },
                "id": "32d9fd8f-91dd-5668-b003-83020ea5439f",
                "manufacturer": Object {
                  "value": "Saregama",
                  "visible": true,
                },
                "name": Object {
                  "value": "Radio",
                  "visible": true,
                },
                "price": Object {
                  "value": 1200,
                  "visible": true,
                },
                "quantity": Object {
                  "value": 10,
                  "visible": true,
                },
                "views": 4,
              },
            ]
          }
          removeFromDeleteList={[Function]}
        />
      </li>
    </ul>
  </div>
</div>
`;
